astropy/nddata/nddata_base.py

```diff
@@ -89,4 +89,12 @@ def uncertainty(self, value):
 
                 raise TypeError('Uncertainty must have attribute '
                                 'uncertainty_type whose type is string.')
+            #Try setting the parent for the uncertainty. This is currently an
+            #enforced requirement for `StdDevUncertainty` for error propagation.
+            try:
+                value.parent_nddata = self
+            except:
```

Comment:
Sorry wasn't clear about what I meant here. It is the except line itself that should have the name of the exception, like this:

```python
except AttributeError:
    pass
```

That way the *only* error this `except` allows to go without comment is an `AttributeError`; any other error generated by the line `value.parent_nddata = self` will (and should) be raised.

In this case it is a little hard to see what other error would be raised, but you could imagine someone subclassing `NDData`, making a setter for `uncertainty` and having a mistake in their code that generate some other error. We don't want to catch that error here, we just want an `AttributeError` raised if `value` has no attribute `parent_nddata`.